torch/lib/THPP/tensors/generic/THSTensor.cpp

```diff
@@ -7,9 +7,9 @@
 #define const_storage_cast(storage) \
   dynamic_cast<const THStorage<real>&>(storage)
 #define const_long_cast(tensor) \
-  dynamic_cast<const THSTensor<long>&>(tensor)
+  dynamic_cast<const THSTensor<int64_t>&>(tensor)
```

Comment:
I think this is inconsistent with the rest of the PR, which uses `longlong` for `int64_t`. Might be better to make it consistent, by changing the rest to be named after `long` instead of `longlong`?